File: tests/expectations/fns.java
import com.sun.jna.*;
import com.sun.jna.ptr.*;

enum BindingsSingleton {
  INSTANCE;
  final Bindings lib = Native.load("", Bindings.class);
}

interface Bindings extends Library {
  Bindings INSTANCE = BindingsSingleton.INSTANCE.lib;


  @Structure.FieldOrder({"noArgs",
                         "anonymousArg",
                         "returnsNumber",
                         "namedArgs",
                         "namedArgsWildcards"})
  class Fns extends Structure implements Structure.ByValue {
    public Fns() {
      super();
    }

    public Fns(Pointer p) {
      super(p);
    }

    public Callback noArgs;
    public Callback anonymousArg;
    public Callback returnsNumber;
    public Callback namedArgs;
    public Callback namedArgsWildcards;

  }

  @Structure.FieldOrder({"noArgs",
                         "anonymousArg",
                         "returnsNumber",
                         "namedArgs",
                         "namedArgsWildcards"})
  class FnsByReference extends Structure implements Structure.ByReference {
    public FnsByReference() {
      super();
    }

    public FnsByReference(Pointer p) {
      super(p);
    }

    public Callback noArgs;
    public Callback anonymousArg;
    public Callback returnsNumber;
    public Callback namedArgs;
    public Callback namedArgsWildcards;

  }


  void root(Fns _fns);

  void no_return();

}

File: tests/expectations/fns.zig
const std = @import("std");

pub const Fns = extern struct {
   _noArgs: ?fn() anyopaque,
   _anonymousArg: ?fn() anyopaque,
   _returnsNumber: ?fn() i32,
   _namedArgs: ?fn(first: i32, snd: i16) i8,
   _namedArgsWildcards: ?fn(_: i32, named: i16, _1: i64) i8,
};

extern fn root(_fns: Fns) anyopaque;

extern fn no_return() anyopaque;
